pkg/controller/infinispan/autoscale.go

```diff
@@ -0,0 +1,204 @@
+package infinispan
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	infinispanv1 "github.com/infinispan/infinispan-operator/pkg/apis/infinispan/v1"
+	"github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	consts "github.com/infinispan/infinispan-operator/pkg/controller/constants"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var autoscaleThreadPool = struct {
+	sync.RWMutex
+	m map[types.NamespacedName]int
+}{m: make(map[types.NamespacedName]int)}
+
+// Starting a go routine that does polling autoscaling on an Infinispan cluster
+func addAutoscalingEquipment(clusterNsn types.NamespacedName, r *ReconcileInfinispan) {
+	autoscaleThreadPool.Lock()
+	_, ok := autoscaleThreadPool.m[clusterNsn]
+	autoscaleThreadPool.Unlock()
+	if !ok {
+		autoscaleThreadPool.Lock()
+		autoscaleThreadPool.m[clusterNsn] = 1
+		autoscaleThreadPool.Unlock()
+		go autoscalerLoop(clusterNsn, r)
+	}
+}
+
+var defaultMinimumPollPeriod = time.Second
+
+func autoscalerLoop(clusterNsn types.NamespacedName, r *ReconcileInfinispan) {
+	log.Info(fmt.Sprintf("Starting loop for autoscaling on cluster %v", clusterNsn))
+	for true {
```

Comment:
Still worries about to heavy load on this loop side
The initial loop must be as lighter as possible, and pass huge work to the separate thread for processing

Reply:
But we could check all cases with manual tests later and detect all possibilities.

Reply:
this is a tradeoff between simplicity and concurrency. I'm not sure the loop is using a lot of resources, most of the time is waiting for response from the net.

Reply:
Yes, as core logic is implemented, always can change it if problems will occurs.